app/bundles/ChannelBundle/Event/ChannelEvent.php

```diff
@@ -71,8 +71,10 @@ public function getRepositoryName($channel)
 
         // if not defined, try the classic naming convention
         $channel = ucfirst($channel);
+        $class   = "\Mautic\\{$channel}Bundle\Entity\\{$channel}";
+        $repo    = new $class();
```

Comment:
But why do we instantiate the class and not passing the FQCN directly?

Reply:
The functions that calls this function expects the result to be class name not a class instance

Reply:
I tried this before, it should return `class-literal` like this 

```
return "\Mautic\\{$channel}Bundle\Entity\\{$channel}::class";
```

Put this syntax does not work

Reply:
Is it PHPSTAN that is causing the trouble?
How about
```php
$class = "\Mautic\\{$channel}Bundle\Entity\\{$channel}";
\assert(class_exists($class));

return $class;
```

Reply:
Brilliant, that worked check commit b18d35038222be9d3ef0cedfeb5fa9223d4fc696